packages/back-end/src/services/dataPipeline.ts

```diff
@@ -0,0 +1,163 @@
+import { FactTableMap } from "back-end/src/models/FactTableModel";
+import { IncrementalRefreshInterface } from "back-end/src/validators/incremental-refresh";
+import { OrganizationInterface } from "back-end/types/organization";
+import { orgHasPremiumFeature } from "back-end/src/enterprise";
+import { SourceIntegrationInterface } from "back-end/src/types/Integration";
+import { ExperimentInterface } from "back-end/types/experiment";
+import {
+  getExperimentSettingsHashForIncrementalRefresh,
+  getMetricSettingsHashForIncrementalRefresh,
+} from "./experimentTimeSeries";
+
+// If the given settings / experiment is not compatible with incremental refresh, throw an error.
+// Otherwise, return void.
+export async function validateIncrementalPipeline({
+  org,
+  integration,
+  snapshotSettings,
+  metricMap,
+  factTableMap,
+  experiment,
+  incrementalRefreshModel,
+  fullRefresh,
+}: {
+  org: OrganizationInterface;
+  integration: SourceIntegrationInterface;
+  snapshotSettings: ExperimentSnapshotSettings;
+  metricMap: Map<string, ExperimentMetricInterface>;
+  factTableMap: FactTableMap;
+  experiment: ExperimentInterface;
+  incrementalRefreshModel: IncrementalRefreshInterface | null;
+  fullRefresh: boolean;
+}): Promise<void> {
+  if (snapshotSettings.skipPartialData) {
+    throw new Error(
+      "'Exclude In-Progress Conversions' is not supported for incremental refresh queries while in beta. Please select 'Include' in the Analysis Settings for Metric Conversion Windows.",
+    );
+  }
+
+  if (!integration.getSourceProperties().hasIncrementalRefresh) {
+    throw new Error("Integration does not support incremental refresh queries");
+  }
+
+  // Check if organization has the incremental refresh feature
+  const hasIncrementalRefreshFeature = orgHasPremiumFeature(
+    org,
+    "incremental-refresh",
+  );
+  if (!hasIncrementalRefreshFeature) {
+    throw new Error(
+      "Organization does not have access to incremental refresh feature",
+    );
+  }
+
+  // Check if pipeline mode is set to incremental
+  const settings = integration.datasource.settings;
+  const canRunIncrementalRefreshQueries =
+    settings.pipelineSettings?.mode === "incremental";
+  if (!canRunIncrementalRefreshQueries) {
+    throw new Error("Integration does not have Pipeline Incremental enabled");
+  }
+
+  if (experiment.activationMetric) {
+    throw new Error(
+      "Activation metrics are not supported for incremental refresh while in beta.",
+    );
+  }
+
+  if (
+    (settings.pipelineSettings?.includedExperimentIds !== undefined &&
+      !settings.pipelineSettings?.includedExperimentIds.includes(
+        experiment.id,
+      )) ||
+    (settings.pipelineSettings?.excludedExperimentIds !== undefined &&
+      settings.pipelineSettings?.excludedExperimentIds.includes(experiment.id))
+  ) {
+    throw new Error(
+      "Experiment is not included in the Pipeline Incremental scope",
```

Comment:
This is a bit cryptic, but I guess this error shouldn't really ever be shown unless we made a mistake.